clients/src/test/java/org/apache/kafka/common/record/DefaultRecordTest.java

```diff
@@ -270,7 +270,7 @@ public void testInvalidNumHeadersPartial() throws IOException {
         DefaultRecord.readPartiallyFrom(inputStream, skipArray, 0L, 0L, RecordBatch.NO_SEQUENCE, null);
     }
 
-    @Test(expected = StringIndexOutOfBoundsException.class)
+    @Test(expected = InvalidRecordException.class)
```

Comment:
Since we parse the header lazily now, what is causing this exception?

Reply:
The exception is caused by ```headerKeyBuffer.limit(headerKeySize);```. The new limit is larger than capacity